sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1093,21 +1126,9 @@ class Dataset[T] private[sql](
     // Otherwise, find the trivially true predicates and automatically resolves them to both sides.
     // By the time we get here, since we have already run analysis, all attributes should've been
     // resolved and become AttributeReference.
-    val cond = plan.condition.map { _.transform {
-      case catalyst.expressions.EqualTo(a: AttributeReference, b: AttributeReference)
-          if a.sameRef(b) =>
-        catalyst.expressions.EqualTo(
-          withPlan(plan.left).resolve(a.name),
```

Comment:
the existing code uses `resolve` not `resolveQuoted`, shall we keep it?

Reply:
ok if you like, but I don't think it will produce any different and resolveQuoted have a better log when there is a error